shell/common/gin_helper/event_emitter_caller.h

```diff
@@ -34,7 +35,15 @@ v8::Local<v8::Value> EmitEvent(v8::Isolate* isolate,
   internal::ValueVector concatenated_args = {gin::StringToV8(isolate, name)};
   concatenated_args.reserve(1 + args.size());
   concatenated_args.insert(concatenated_args.end(), args.begin(), args.end());
-  return internal::CallMethodWithArgs(isolate, obj, "emit", &concatenated_args);
+
+  v8::TryCatch try_catch(isolate);
+  v8::Local<v8::Value> value =
+      internal::CallMethodWithArgs(isolate, obj, "emit", &concatenated_args);
+  if (try_catch.HasCaught()) {
+    LOG(ERROR) << "An unknown error occurred in the event emitter";
+  }
+
+  return value;
```

Comment:
Is returning the value after an error a good best practice here?

Reply:
Yes, `CallMethodWithArgs` guarantees the value is non-empty even if an error is thrown.